### DISCUSSION TITLE
Parallel route props

### DISCUSSION THREAD

**Discussion by plevamatus** [vercel/next.js#50368] [App Router]
I would like to implement side navigation (@sideNav) as parallel route, but I want to control its visibility by button which is outside this component. To do this, I have to keep information about current state in layout.tsx and pass it down as prop to @sideNav. Is it possible to do it like this?

**Comment by devjiwonchoi** [comment]
Not sure what you are trying to achieve, but try to set your `layout.tsx` as [Client Component](https://nextjs.org/docs/getting-started/react-essentials#client-components) by setting `use client` on the top of the file and use Client-Side React Hooks such as `useEffect` to pass the state to @sidenav as you want.

**Comment by veih14** [comment]
I have the same question. Is this possible to do?

**Comment by logemann** [comment]
Also facing the issue that i need to pass props to my tabs. How to do that?
 Instead of {tab1} i need to have something like <Tab1 ...props />

```
const tabs = ["Tab1", "Tab2", "Tab3", "Tab4"]

export default function layout({
                                               tab1,
                                               tab2,
                                               tab3,
                                               tab4
                                           }: {
    tab1: React.ReactNode,
    tab2: React.ReactNode,
    tab3: React.ReactNode,
    tab4: React.ReactNode
}) {
   
    return (
            <>
                <TabGroup>
                    <TabList className="space-x-6 bg-primary-50 px-4 py-2 rounded-md">
                        {tabs.map(tab => <Tab key={tab}
                                              className="data-[selected]:bg-primary-100 data-[selected]:text-primary-900
                                                    text-primary-800 outline-0 rounded-md px-3 py-2 text-sm font-medium">
                            {tab}
                        </Tab>)}
                    </TabList>
                    <TabPanels className="mt-8">
                        {tab1}
                        {tab2}
                        {tab3}
                        {tab4}
                    </TabPanels>
                </TabGroup>
        </>
);
};
```

**Comment by tiavina-mika** [comment]
More than 1 years later, it's not possible I guess?
